### PR TITLE
[core] More GcsClient Accessor methods.

### PR DESCRIPTION
**Pull request by rynewang**
Add `num_deleted` to InternalKVAccessor::Del and [Async]CheckAlive to  NodeInfoAccessor. This is needed for feature parity with PythonGcsClient.

Added with C++ Tests.

### CODE REVIEW

File: src/ray/gcs/gcs_client/accessor.cc [hunk 1]
```diff
@@ -491,22 +491,47 @@ Status NodeInfoAccessor::AsyncRegister(const rpc::GcsNodeInfo &node_info,
...
+
+  return AsyncCheckAlive(
+      raylet_addresses,
+      timeout_ms,
+      [callback](const Status &status, const std::vector<bool> &nodes_alive) {
+        if (!status.ok()) {
+          callback(status, false);
+          return;
+        } else {
+          RAY_CHECK(nodes_alive.size() == 1);
```
Comment (jjyao) [inline src/ray/gcs/gcs_client/accessor.cc:506]: RAY_CHECK_EQ

File: src/ray/gcs/gcs_client/accessor.cc [hunk 2]
```diff
@@ -491,22 +491,47 @@ Status NodeInfoAccessor::AsyncRegister(const rpc::GcsNodeInfo &node_info,
...
+  }
+  size_t num_raylets = raylet_addresses.size();
   client_impl_->GetGcsRpcClient().CheckAlive(
       request,
-      [callback](auto status, const auto &reply) {
+      [num_raylets, callback](const Status &status, const rpc::CheckAliveReply &reply) {
         if (status.ok()) {
-          RAY_CHECK(reply.raylet_alive().size() == 1);
-          callback(status, reply.raylet_alive()[0]);
+          RAY_CHECK(static_cast<size_t>(reply.raylet_alive().size()) == num_raylets);
```
Comment (jjyao) [inline src/ray/gcs/gcs_client/accessor.cc:524]: RAY_CHECK_EQ

File: src/ray/gcs/gcs_client/accessor.cc [hunk 3]
```diff
@@ -491,22 +491,47 @@ Status NodeInfoAccessor::AsyncRegister(const rpc::GcsNodeInfo &node_info,
...
+          return;
+        } else {
+          RAY_CHECK(nodes_alive.size() == 1);
+          callback(status, nodes_alive[0]);
+        }
+      });
+}
+
+Status NodeInfoAccessor::AsyncCheckAlive(const std::vector<std::string> &raylet_addresses,
+                                         int64_t timeout_ms,
```
Comment (jjyao) [inline src/ray/gcs/gcs_client/accessor.cc:513]: timeout_ms is not used?

**Review by jjyao — APPROVED** [review]

File: src/ray/gcs/gcs_client/accessor.cc (hunk 1 above)
Comment (rynewang) [reply src/ray/gcs/gcs_client/accessor.cc]: done

File: src/ray/gcs/gcs_client/accessor.cc (hunk 2 above)
Comment (rynewang) [reply src/ray/gcs/gcs_client/accessor.cc]: done

File: src/ray/gcs/gcs_client/accessor.cc (hunk 3 above)
Comment (rynewang) [reply src/ray/gcs/gcs_client/accessor.cc:513]: done

**Comment by jjyao** [comment]
Merge conflicts

**Pull request merged**